### PR TITLE
WINDUP-666 (update condition + new rule): Add support for query on XML…

### PR DESCRIPTION
**Pull request by mbriskar**
… files by dtd namespace

### CODE REVIEW

File: rules-xml/api/src/main/java/org/jboss/windup/rules/apps/xml/condition/XmlFile.java [hunk 1]
```diff
@@ -358,15 +365,22 @@ else if (iterated instanceof XmlFileModel)
...
                     {
                         continue;
                     }
-                    else if (xpathString == null)
+                    if(publicId != null && ( (doctype.getPublicId() == null) || !doctype.getPublicId().matches(publicId))) {
+                        continue;
+                    }
+                    if(dtdNamespace != null && ( (doctype.getSystemId() == null) || !doctype.getSystemId().matches(dtdNamespace))) {
+                        continue;
+                    }
```
Comment (mbriskar) [inline rules-xml/api/src/main/java/org/jboss/windup/rules/apps/xml/condition/XmlFile.java]: this is not very nice. I have created separate https://issues.jboss.org/browse/WINDUP-887 for refactoring XmlFile into multiple subconditions

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/759/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1387/

File: rules-xml/api/src/main/java/org/jboss/windup/rules/apps/xml/condition/XmlFile.java (hunk 1 above)
Comment (jsight) [inline rules-xml/api/src/main/java/org/jboss/windup/rules/apps/xml/condition/XmlFile.java]: Good idea.

**Comment by mbriskar** [comment]
I added new extra commit with handles xsd and new PR in windup-rulesets as well ( https://github.com/windup/windup-rulesets/pull/107)

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-pr-builder/1388/

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://wonka.mw.lab.eng.bos.redhat.com/jenkins//job/windup-core-PR-dependents-check/760/

**Pull request merged**